File: client-domain/src/main/java/com/jraska/github/client/users/UsersRepository.java
package com.jraska.github.client.users;

import java.util.List;

import io.reactivex.Observable;
import io.reactivex.Single;

public interface UsersRepository {
  Single<List<User>> getUsers(int since);

  Observable<UserDetail> getUserDetail(String login, int reposInSection);

  Observable<RepoDetail> getRepoDetail(String owner, String repoName);
}

File: feature-users/src/main/java/com/jraska/github/client/users/UsersRepository.kt
package com.jraska.github.client.users

import io.reactivex.Observable
import io.reactivex.Single

interface UsersRepository {
  fun getUsers(since: Int): Single<List<User>>

  fun getUserDetail(login: String, reposInSection: Int): Observable<UserDetail>

  fun getRepoDetail(owner: String, repoName: String): Observable<RepoDetail>
}
